src/main/java/org/isf/menu/service/MenuIoOperations.java

```diff
@@ -362,25 +387,28 @@ public UserGroup newUserGroup(UserGroup userGroup, List<Permission> permissions)
 
 	/**
 	 * Updates an existing {@link UserGroup} in the DB
-	 * 
 	 * @param aGroup - the {@link UserGroup} to update
-	 * @return {@code true} if the group has been updated, {@code false} otherwise.
+	 * @return the {@link UserGroup} that has been updated
 	 * @throws OHServiceException When failed to update the user group
 	 */
-	public boolean updateUserGroup(UserGroup aGroup) throws OHServiceException {
-		return groupRepository.updateDescription(aGroup.getDesc(), aGroup.getCode()) > 0;
+	public UserGroup updateUserGroup(UserGroup aGroup) throws OHServiceException {
+		groupRepository.update(aGroup.getDesc(), aGroup.isDeleted(), aGroup.getCode());
+
+		return findByCode(aGroup.getCode());
 	}
 
 	/**
-	 * Updates an existing {@link UserGroup} and the related permissions
-	 * If permissions list is empty, the existing permissions are kept,
-	 * otherwise they're replaced with the provided ones.
-	 *
+	 * Updates an existing {@link UserGroup} and the related permissions If permissions list is empty, the existing permissions are kept, otherwise they're
+	 * replaced with the provided ones.
 	 * @param userGroup - the {@link UserGroup} to update
-	 * @return {@code true} if the group has been updated, {@code false} otherwise.
+	 * @return the {@link UserGroup} that has been updated
 	 * @throws OHServiceException When failed to update user group
 	 */
-	public boolean updateUserGroup(UserGroup userGroup, List<Permission> permissions) throws OHServiceException {
+	public UserGroup updateUserGroup(UserGroup userGroup, List<Permission> permissions) throws OHServiceException {
+		UserGroup group = findByCode(userGroup.getCode(), true);
+		if (group.isDeleted() && userGroup.isDeleted()) {
+			throw new OHServiceException(new OHExceptionMessage("angal.common.denied.msg"));
```

Comment:
if the above proposal is accepted, this could be again ""angal.userbrowser.alreadysoftdeleted.msg = The user is already set as deleted."

```suggestion
			throw new OHServiceException(new OHExceptionMessage(MessageBundle.getMessage("angal.userbrowser.alreadysoftdeleted.msg")));
```